### PR TITLE
New PTF implementation and --dry-run enablement

### PR DESCRIPTION
**Pull request by nfelt14**

### CODE REVIEW

File: behave/runner.py [hunk 1]
```diff
@@ -906,7 +933,7 @@ def load_step_definitions(self, extra_step_paths=[]):
                     if os.path.isfile(os.path.join(root_dir, temp_scope_pi_step_import)):
                         scope_step_pi_import = temp_scope_pi_step_import
                     else:
-                        scope_step_pi_import = "devices/scopes/series_5/mso/pi/all_steps.py"
+                        scope_step_pi_import = "python_test_framework/bdd/devices/scopes/series_5/mso/pi/all_steps.py"
```
Comment (nlynchjo) [inline behave/runner.py]: oh I can see how that would be a problem now!

File: behave/runner.py [hunk 2]
```diff
@@ -971,8 +998,8 @@ def load_step_definitions(self, extra_step_paths=[]):
                                               "and revision.")
 
             else:
-                raise NotImplementedError("Congrats, you magically got an unsupported device ({0}) through the config "
-                                          "parser, now go tell Joshua Sleeper or Jonathan David Ice.".format(dev_name))
+                raise AssertionError("No valid device was specified in the config file, find your local SQE and talk "
+                                     "with them about how to fix it.")
```
Comment (nlynchjo) [inline behave/runner.py]: 👏

File: behave/runner.py [hunk 3]
```diff
@@ -889,15 +887,44 @@ def load_step_definitions(self, extra_step_paths=[]):
...
-
-        for dev_name, (_, series, dev_type, revision, form_factor, _) in devices.items():
+        if self.config.dry_run:
+            # [dev_name, (_, series, dev_type, revision, form_factor, _)]
+            try:
+                device_list = [('scope', (None, self.config.userdata['dry_run_scope_series'],
+                                          self.config.userdata['dry_run_scope_type'], '', None, None)),
+                               ('AFG', (None, self.config.userdata['dry_run_afg_series'],
+                                        self.config.userdata['dry_run_afg_type'], '', None, None)),
+                               ('AWG', (None, self.config.userdata['dry_run_awg_series'],
```
Comment (nlynchjo) [inline behave/runner.py]: worth it to add PROBE_SWITCHER here too?

Comment (nfelt14) [reply behave/runner.py]: Since we have no steps for them I don't think it is necessary.

**Pull request merged**